src/main/scala/org/bitcoins/core/util/BitcoinScriptUtil.scala

```diff
@@ -300,6 +308,12 @@ trait BitcoinScriptUtil {
     }
     return true
   }
+
+  def correctScriptNumberRepresentation(num : ScriptNumber) : Option[ScriptOperation] = {
```

Comment:
This looks a little hairy. I thought about this more and I think we should change the name to `minimalScriptNumberRepresentation` as that is a little less ambiguous compared to `correctScriptNumberRepresentation`. Also, I think we can just return a `ScriptNumber` instead of an `Option[ScriptOperation]`. We can just assume this function returns `num` if it is the minimally encoded number, or else it transforms `num` into the appropriate `ScriptNumberOperation`. [Technically, in our type hierarchy, `ScriptNumberOperation` extends ScriptNumbe](https://github.com/bitcoin-s/bitcoin-s-core/blob/master/src/main/scala/org/bitcoins/core/script/constant/Constants.scala#L211)r, so we can get away with just returning a ScriptNumber.  I can help with this if needed.